data/src/config/buffer.rs

```diff
@@ -40,6 +44,8 @@ impl ServerMessages {
             source::server::Kind::Part => Some(self.part),
             source::server::Kind::Quit => Some(self.quit),
             source::server::Kind::Join => Some(self.join),
+            source::server::Kind::StatusSuccess => Some(self.status_success),
+            source::server::Kind::StatusError => Some(self.status_error),
```

Comment:
Let's add a new method `get_internal` which matches on `source::Internal` since that's what these messages actually are. Then we can get rid of this unneeded server source.

Reply:
Alternatively, we can add a new `InternalMessages` / `buffer.internal_messages.success / error` since that better maps to our data model and better describes what these messages are. They aren't coming from the upstream server.

Or maybe we rename `server_messages` -> `messages` so it can encapsulate both. (we can alias `messages` to `server_messages` during deserialization as to not break backwards compatibility).

@casperstorm thoughts?

Reply:
I like the explicit `InternalMessages` / `buffer.internal_messages.success / error`. I had considered originally doing something like `buffer.server_messages.status.error`. So I like the internal_messages approach you laid out here!

Reply:
Refactored! This was how I should have written it in the first place. Hope it's this is more acceptable.

Reply:
I think `internal_messages` works well.